File: src/main/kotlin/net/println/kotlin/Chapter10/io/IO.java
package net.println.kotlin.Chapter10.io;

import java.io.*;

public class IO {
    public static void main(String... args) {
        BufferedReader bufferedReader = null;
        try {
            bufferedReader = new BufferedReader(new FileReader(new File("build.gradle")));
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                bufferedReader.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: src/main/kotlin/net/println/kotlin/Chapter10/io/IO.kt
package net.println.kotlin.Chapter10.io

import java.io.BufferedReader
import java.io.File
import java.io.FileReader

fun main(args: Array<String>) {
    val file = File("build.gradle")
    val bufferedReader = BufferedReader(FileReader(file))
    var line:String

    while (true) {
        line = bufferedReader.readLine()?:break
        println(line)

    }
    bufferedReader.close()
}